user.go

```diff
@@ -57,38 +63,76 @@ func (u *User) Defaults(ctx context.Context) {
 		t := Now()
 		u.UpdatedAt = &t
 	}
+
+	if !u.EmailVerified {
+		u.EmailToken = zhttp.SecretP()
+	}
 }
 
 // Validate the object.
-func (u *User) Validate(ctx context.Context) error {
+func (u *User) Validate(ctx context.Context, validatePassword bool) error {
 	v := zvalidate.New()
 
 	v.Required("site", u.Site)
-	v.Required("name", u.Name)
 	v.Required("email", u.Email)
-
-	v.Len("name", u.Name, 1, 200)
 	v.Len("email", u.Email, 5, 255)
 	v.Email("email", u.Email)
+	if u.Name != "" {
+		v.Len("name", u.Name, 1, 200)
+	}
+
+	if validatePassword {
+		sp := string(u.Password)
+		v.Required("password", u.Password)
+		v.UTF8("password", sp)
+		v.Len("password", sp, 8, 30)
```

Comment:
@arp242 Just out of curiosity, why are passwords limited to 30 characters?

Reply:
It's basically a hack so I can detect if the `Password` field is the hashed of plaintext version: https://github.com/zgoat/goatcounter/blob/e865135/user.go#L95-L99

This condition should never be triggered; it's just a safety check.

Reply:
interesting 😀I've got my password manager set to generate 48 characters long passwords, that's why I noticed^^

Reply:
Right; that's quite long. The stored hash is actually 60 bytes, so no reason to increase the limit to 50. I changed it in d445e45; will deploy in a bit.

Reply:
Ooh nice 😍🙏